rust/src/treefile.rs

```diff
@@ -178,18 +178,11 @@ fn take_archful_pkgs(
     Ok(archful_pkgs)
 }
 
-// If a passwd/group file is provided explicitly, load it as a fd
-fn load_passwd_file<P: AsRef<Path>>(
-    basedir: P,
-    v: &Option<CheckPasswd>,
-) -> Result<Option<fs::File>> {
-    if let Some(ref v) = *v {
-        let basedir = basedir.as_ref();
-        if let Some(ref path) = v.filename {
-            return Ok(Some(utils::open_file(basedir.join(path))?));
-        }
-    }
-    Ok(None)
+/// If a passwd/group file is provided explicitly, load it as a fd.
+fn load_passwd_file<P: AsRef<Path>>(basedir: P, cfg: &CheckFile) -> Result<Option<fs::File>> {
```

Comment:
Feels a bit weird to have this function now always return `Some`.  Maybe do the `Some()` wrapping in the callers instead?